pkg/collect/host_filesystem_performance.go

```diff
@@ -90,3 +98,324 @@ func (f FSPerfResults) String() string {
+	P70   int `json:"70.000000,omitempty"`
+	P80   int `json:"80.000000,omitempty"`
+	P90   int `json:"90.000000,omitempty"`
+	P95   int `json:"95.000000,omitempty"`
+	P99   int `json:"99.000000,omitempty"`
+	P995  int `json:"99.500000,omitempty"`
+	P999  int `json:"99.900000,omitempty"`
+	P9995 int `json:"99.950000,omitempty"`
+	P9999 int `json:"99.990000,omitempty"`
+}
+
+func (d FioDiskUtil) String() string {
+	//Disk stats (read/write):
+	//rbd4: ios=30022/11982, merge=0/313, ticks=1028675/1022768, in_queue=2063740, util=99.67%
+	var du string
+	du += fmt.Sprintf("  %s: ios=%d/%d merge=%d/%d ticks=%d/%d in_queue=%d, util=%f%%", d.Name, d.ReadIos,
+		d.WriteIos, d.ReadMerges, d.WriteMerges, d.ReadTicks, d.WriteTicks, d.InQueue, d.Util)
+	return du
+}
+
+func parseCollectorOptions(hostCollector *troubleshootv1beta2.FilesystemPerformance) ([]string, *FioJobOptions, error) {
+
+	var operationSize uint64 = 1024
+	if hostCollector.OperationSizeBytes > 0 {
+		operationSize = hostCollector.OperationSizeBytes
+	}
+	var fileSize uint64 = 10 * 1024 * 1024
+	if hostCollector.FileSize != "" {
+		quantity, err := resource.ParseQuantity(hostCollector.FileSize)
+		if err != nil {
+			return nil, nil, errors.Wrapf(err, "failed to parse fileSize %q", hostCollector.FileSize)
+		}
+		fileSizeInt64, ok := quantity.AsInt64()
+		if !ok {
+			return nil, nil, errors.Wrapf(err, "failed to parse fileSize %q", hostCollector.FileSize)
+		}
+		if fileSizeInt64 <= 0 {
+			return nil, nil, errors.Wrapf(err, "fileSize %q must be greater than 0", hostCollector.FileSize)
+		}
+		fileSize = uint64(fileSizeInt64)
+	}
+
+	if hostCollector.Directory == "" {
+		return nil, nil, errors.New("Directory is required to collect filesystem performance info")
+	}
+
+	latencyBenchmarkOptions := FioJobOptions{
+		RW:        "write",
+		IOEngine:  "sync",
+		FDataSync: "1",
+		Directory: hostCollector.Directory,
+		Size:      strconv.FormatUint(fileSize, 10),
+		BS:        strconv.FormatUint(operationSize, 10),
+		Name:      "fsperf",
+		RunTime:   "120",
+	}
+
+	command := buildFioCommand(latencyBenchmarkOptions)
+
+	return command, &latencyBenchmarkOptions, nil
+}
+
+func buildFioCommand(opts FioJobOptions) []string {
+	command := []string{"fio"}
+	v := reflect.ValueOf(opts)
+	t := reflect.TypeOf(opts)
+	for i := 0; i < v.NumField(); i++ {
+		field := t.Field(i)
+		value := v.Field(i)
+		if !value.IsZero() {
+			command = append(command, fmt.Sprintf("--%s=%v", strings.ToLower(field.Name), value.Interface()))
+		}
+	}
+	command = append(command, "--output-format=json")
+	return command
+}
+
+func collectFioResults(hostCollector *troubleshootv1beta2.FilesystemPerformance) (*FioResult, error) {
+
+	command, opts, err := parseCollectorOptions(hostCollector)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to parse collector options")
+	}
+
+	klog.V(2).Infof("collecting fio results: %s", strings.Join(command, " "))
+	output, err := exec.Command(command[0], command[1:]...).Output()
```

Comment:
Lets use `exec.CommandContext` command so that we can respect the collector's [timeout](https://troubleshoot.sh/docs/host-collect-analyze/filesystemPerformance/#timeout-required). You'll end up having something like below after having passed in the context to `collectFioResults()` which is created in [here](https://github.com/replicatedhq/troubleshoot/blob/fd29012624ad8167ec87b16108407048ae38484d/pkg/collect/host_filesystem_performance_linux.go#L50)

```go
output, err := exec.CommandContext(ctx, command[0], command[1:]...).Output()
```